control-plane/src/modules/router.ts

```diff
@@ -348,4 +350,39 @@ export const router = s.router(contract, {
       body: result,
     };
   },
+  getDeploymentUploadDetails: async (request) => {
+    if (!UPLOAD_BUCKET) {
+      return {
+        status: 501,
+      };
+    }
+
+    const owner = await auth.jobOwnerHash(request.headers.authorization);
```

Comment:
Currently a _regular_ cluster token (with `cloudEnabled` = true) is all that is required.
We will probably want some more granular ACL?